src/directives/leaflet.js

```diff
@@ -24,9 +23,9 @@ angular.module("leaflet-directive", []).directive('leaflet',
         transclude: true,
         template: '<div class="angular-leaflet-map"><div ng-transclude></div></div>',
         controller: function ($scope) {
-            _leafletMap = $q.defer();
+            this._leafletMap = $q.defer();
```

Comment:
Yes which is why I said the link function will make it unique. Since your using this, so yes this is correct. I just was looking for something to reference the angular clean up code you speak of.